File: Crio/crio_programming_interview_problems-master/SortList/SortList.cpp
#include "../crio/cpp/io/FastIO.hpp"
#include "../crio/cpp/io/PrintMatrix.hpp"
#include "../crio/cpp/io/ReadMatrix.hpp"
#include <bits/stdc++.h>
#include "Solution.cpp"

using namespace std;

// CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
// CRIO_SOLUTION_END_MODULE_L1_PROBLEMS

int main() {
    FastIO();
    int n;
    cin >> n;
    vector<int> nums;
    ReadMatrix<int>().OneDMatrix(n, nums);
    ListNode* list = ListNode().createList(nums);
    SortList obj;
    ListNode* sortedList = obj.sortList(list);
    vector<int> result = ListNode().extractList(sortedList);
    PrintMatrix<int>().OneDMatrix(result, " ");
    return 0;
}

File: Crio/crio_programming_interview_problems-master/SortList/SortList.py
from Solution import *
from crio.ds.List.ListNode import *

# CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
# CRIO_SOLUTION_END_MODULE_L1_PROBLEMS

def print_list(head):
    while head != None:
        print(head.val, end = " ")
        head = head.next

def main():
   n = int(input().strip())
   if n > 0 :
      lst = [int(x) for x in input().strip().split()]
      lst = createList(lst)
      print_list(sort_list(lst))
   else:
      print_list(sort_list(None))

if __name__ == '__main__':
    main()
